app/main/views/suppliers.py

```diff
@@ -533,16 +536,15 @@ def duns_number():
             # Otherwise the number is good, add the 'primaryName' to the session and continue
             try:
                 organization = direct_plus_client.get_organization_by_duns_number(form.duns_number.data)
-                if organization is not None:
-                    company_name = organization['primaryName']
-            except (KeyError, ValueError):
+                company_name = organization['primaryName']
+            except (DUNSNumberInvalid, DUNSNumberNotFound):
+                form.duns_number.errors = ["DUNS number not found"]
+            except (DirectPlusError, KeyError, ValueError):
```

Comment:
```suggestion
            except (DirectPlusError, KeyError):
```

Is there anything able to raise a `ValueError` any more?

Reply:
Yes, `get_organization_by_duns_number()` can still raise a `ValueError` when calling `response.json()`

https://github.com/alphagov/digitalmarketplace-utils/blob/master/dmutils/direct_plus_client.py#L132

I guess we could propagate it all the way to a 500 if we wanted to though, by removing ValueError.

Reply:
Very true - since it wasn't in the doc comment, I missed that it could still throw `KeyError` and `ValueError`.

In that case, I think the behaviour here is correct. It's more that I would like the Direct+ client to either document those exceptions or turn them into `DirectPlusError`s. Sorry - missed that when reviewing the previous PR.